packages/adapters/solana/src/client.ts

```diff
@@ -1,699 +1,451 @@
-
-// -- Client --------------------------------------------------------------------
-export class SolanaAdapter implements ChainAdapter {
-  private appKit: AppKit | undefined = undefined
-
-  private authProvider?: Provider
-
+export class SolanaAdapter extends AdapterBlueprint {
+  private connectionSettings: Commitment | ConnectionConfig
   private w3mFrameProvider?: W3mFrameProvider
+  private authProvider?: AuthProvider
+  private authSession?: AuthProvider.Session
 
-  public options: CoreOptions | undefined = undefined
-
-  public wallets?: BaseWalletAdapter[]
-
-  public caipNetworks: CaipNetwork[] = []
-
-  public readonly chainNamespace: ChainNamespace = CommonConstantsUtil.CHAIN.SOLANA
-
-  public networkControllerClient?: NetworkControllerClient
-
-  public connectionControllerClient?: ConnectionControllerClient
-
-  public connectionSettings: Commitment | ConnectionConfig
-
-  private availableProviders: Provider[] = []
-
-  private provider: Provider | undefined
-
-  private authSession: AuthProvider.Session | undefined
-
-  public defaultCaipNetwork: CaipNetwork | undefined = undefined
-
-  public readonly adapterType: AdapterType = 'solana'
+  constructor(options: AdapterOptions = {}) {
+    super({})
+    this.namespace = CommonConstantsUtil.CHAIN.SOLANA
+    this.connectionSettings = options.connectionSettings || 'confirmed'
 
-  public constructor(options: AdapterOptions) {
-    const { wallets, connectionSettings = 'confirmed' } = options
-
-    this.wallets = wallets
-    this.connectionSettings = connectionSettings
+    EventsController.subscribe(state => {
+      if (state.data.event === 'SELECT_WALLET') {
+        const isMobile = CoreHelperUtil.isMobile()
+        const isClient = CoreHelperUtil.isClient()
 
-    ChainController.subscribeKey('activeCaipNetwork', caipNetwork => {
-      const caipAddress = this.appKit?.getCaipAddress(this.chainNamespace)
-      const isSolanaAddress = caipAddress?.startsWith('solana:')
-      const isSolanaNetwork = caipNetwork?.chainNamespace === this.chainNamespace
+        if (isMobile && isClient) {
+          if (state.data.properties?.name === 'Phantom' && !('phantom' in window)) {
+            const href = window.location.href
+            const protocol = href.startsWith('https') ? 'https' : 'http'
+            const host = href.split('/')[2]
+            const ref = `${protocol}://${host}`
+            window.location.href = `https://phantom.app/ul/browse/${href}?ref=${ref}`
+          }
 
-      if (caipAddress && isSolanaAddress && isSolanaNetwork) {
-        this.syncAccount({
-          address: CoreHelperUtil.getPlainAddress(caipAddress),
-          caipNetwork
-        })
+          if (state.data.properties?.name === 'Coinbase Wallet' && !('coinbaseSolana' in window)) {
+            const href = window.location.href
+            window.location.href = `https://go.cb-w.com/dapp?cb_url=${href}`
+          }
+        }
```

Comment:
we should maybe abstract these to something like `handleMobileWalletRedirection` or similar